rfcs/0064-documentation-format.md

```diff
@@ -0,0 +1,200 @@
+- A set of requirements for the doc format is decided through the RFC discussion
+- Doc format candidates are collected and evaluated to see if they fulfil the requirements. This should include a small demonstration
+- A short objective overview of each valid candidate format is written, along with their advantages/disadvantages
+- A [Discourse](https://discourse.nixos.org/) post is created with these overviews, along with a poll such that people can vote on the formats they prefer. This poll will be open to the whole community and should be advertised as such
+- With the result of the poll as an input, the shepherd team decides on a doc format to be used. This decision is added to the RFC, after which it is merged.
+- People are then free to work towards that new documentation format and the committers of nixpkgs must not oppose these efforts due to the format choice
+
+## Poll
+
+Since documentation needs to be written and read by the many people from the community, we should incorporate them as possible into this important decision. Discussions in RFCs aren't optimal for this, since not many people are willing to comment, meaning the few people with the strongest opinion will be the main participants, in addition to GitHub's comment section being very annoying to navigate. Because of this, a public poll on Discourse is held instead, allowing many people to easily give their preferences.
+
+The poll is of the following form:
+- Multiple-choice, allowing people to select all formats they agree with
+- Results are only shown when the poll is closed for it to not be influenced by non-final tallies
+- Answers can be changed while the poll is still active, allowing people to discuss about formats and change their opinion (this is not optional in Discourse)
+- It runs for 1 month to give enough time for less-active people to see it
+- Who voted for which options is made public (Only possible with bar chart in Discourse) TODO: Do we want this or not? Why would we?
+
+## Requirements
+
+- Can be converted to HTML and man pages
+- Inter-file references for being able to link to options from anywhere
+- Ability to create link anchors to most places such that we can link to e.g. paragraphs
+- Widespread editor integration featuring at least highlighting and preferably live-view
+- Good error detection in toolchain and editors, e.g. with a fast and good processor
+- Is decently fast to fully generate, in the range of 10 seconds for the full documentation on an average machine
+- Closure-size of toolchain should be [small](https://github.com/NixOS/nixpkgs/issues/63513).
+- Supports syntax highlighting (with Nix support)
+- Active community supporting the tooling infrastructure
+- Good conversion story from Docbook
+- Can be used and integrated from NixOS option declarations
+
+### Nice-to-have's
+
+- Annotations/links inside code listings for e.g. linking to option docs in `configuration.nix` snippets
```

Comment:
How is this different from *Inter-file references for being able to link to options from anywhere*?

Reply:
Shooting from the hip... I think it describes the same behavior from a reader's perspective, but it means the engine is doing more than identifying a code block, a syntax type, and handing it off to the appropriate highlighter. 

You have to parse (or at least partially parse) code blocks at build time, identify the different entities they contain, and use heuristics or direct lookups to resolve those local code entities into absolute paths to the appropriate documentation. 

If the highlighter isn't collaborating in this work, it probably also entails postparsing the highlighted blocks to inject the links, and potentially modifying the stylesheets to make sure the links don't stomp all over the syntax highlighting.

Reply:
Sounds complicated :)

Are there examples of documentation format parsers doing this?